appcues/src/main/java/com/appcues/ui/primitive/OptionSelectPrimitive.kt

```diff
@@ -250,9 +250,16 @@ private fun List<OptionSelectPrimitive.OptionItem>.ComposePicker(
                     expanded = false
                     valueSelectionChanged(setOf(it.value))
                 }) {
-                    it.content.Compose()
+                    it.contentView(selectedValue == it.value).Compose()
                 }
             }
         }
     }
 }
+
+private fun OptionSelectPrimitive.OptionItem.contentView(isSelected: Boolean) =
```

Comment:
this logic already works, a compose way of doing this is to use derivedStateOf to keep the state of "content" based on the selectedState. this way calculation happens only one (I think, we could test this out)

I'm still learning this but from the example I guess we could make it work somewhat like this:

```kotlin

    val contentView by remember(isSelected) {
        derivedStateOf { 
             selectedContent?.let { if(isSelected) it else content } ?: run { content }
         }
    }
```

Reply:
Thanks for the tip, does this capture what you had in mind @andretortolano ? https://github.com/appcues/appcues-android-sdk/pull/230/commits/5604cd59025519e01a48d3df8ac3271201bb7c7f

Reply:
yes!